src/components/pokemon/pokemon.component.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import apiRequests from 'FetchAPI/http.lib';
+import restApiLinks from 'FetchAPI/restful-api.links';
+
+import CustomButton from 'Components/custom-button/custom-button.component';
+
+import './pokemon.styles.scss';
+
+class Pokemon extends React.Component {
+  capitalizeWord = word => {
+    const newWord = word.split('')[0].toUpperCase() + word.slice(1);
+    return newWord;
+  };
+
+  catchHandler = async pokemon => {
+    const backRequest = await apiRequests.post(
+      restApiLinks.capturedPokemons,
+      pokemon
+    );
+    const { data, response } = await backRequest;
+    if (response.ok) {
+      await apiRequests.delete(`${restApiLinks.freePokemons}/${pokemon.id}`);
+      this.props.catchPokemon(data);
+    }
+  };
+
+  rescueHandler = async pokemon => {
+    const backRequest = await apiRequests.post(

```

Comment:
let's create `pokemonsApiService` that provides methods like
 `pokemonsApiService.catchPokemon: (pokemon) => {`
`return apiRequests.post(restApiLinks.freePokemons, pokemon`
`);`
 and so on. It will let you separate http requests configuration from business logic of your app (and if it will be needed, change transport protocol easy)

Reply:
Will do that, it feels like a way better approach

Reply:
and more - don't you think that all this requests should happen in sagas? - result of  this async requests affects state of your app. Here you need just to make action like `requestMakePokemonFree(pokemon)` and handle all api interactions in sagas

Reply:
Actually I have tried to do such logic in sagas but met a problem: I can't find a way to bring pokemon data in sagas file. Probably I should use Redux State and `currentPokemon` field somehow... That's my guess

Reply:
you can just send pokemon data with common redux action - something like
`export function* watchRequestUserLogin() {`
    `yield takeEvery(REQUEST_LOGIN_USER, fetchUserLogin);`
}`

`function* fetchUserLogin(action) {`
    `const { email, password } = action.payload;`
    `try {`
        `const data = yield call(() => userService.login(email, password));`
        `yield put(receiveLoginUserInfo(data));`
        `yield put(push('/'));`
   ` }`
   ` catch (error) {`
        `console.log(error.message);`
        `yield put(loginUserFail());`
   ` }`
`}`

Reply:
in that example all what you'll do in component - just dispatch simple redux action
`onClickLogin= () => dispatch({type: REQUEST_LOGIN_USER, payload: loginData});`
(of course you shouldn't use dispatch directly - I wrote it just for simpler explanation :)